get-artifacts/get_artifacts.sh

```diff
@@ -0,0 +1,178 @@
+#!/bin/bash
+
+# fast fail.
+set -eo pipefail
+
+function echoerr() {
+  cat <<< "$@" 1>&2;
+}
+
+# Check prerequisites.
+function check_command() {
+  for var in "$@"; do
+    if ! (command -v "$var" >/dev/null 2>&1); then
+      echoerr "This command requires $var to be installed"
+      exit 1
+    fi
+  done
+}
+check_command jq curl zip echo getopts
+
+function usage() {
+  echo -t token used to communicate with github actions.
+  echo -h number of workflow_runs to pull artifacts from.
+  echo -a space seperated list of artifact names.
+  echo -b optional branch to pull historical artifacts from.
+  echo -w workflow file name of of workflow to pull artifact from.
+  echo -d target directory where subdirectories for jobs will be created and artifacts will be unzip to.
+  echo -i the single workflow_run from which to download artifacts.  Overrides -b, -w, and -h
+  echo -z decompress and delete the original downloaded artifacts.
+  echo -? this message.
+  echo output will files will be written to the -t target directory.
+}
+
+TOKEN=
+HISTORY=
+ARTIFACTS=
+REPO=
+WORKFLOW=
+BRANCH=
+TARGET_DIR=
+WORKFLOW_RUN_ID=
+DECOMPRESS=false
+
+
+while getopts 'h:a:r:w:b:i:t:d:z' OPTION; do
+  case "$OPTION" in
+    h)
+      HISTORY="$OPTARG"
+      ;;
+    a)
+      IFS=' ' read -ra ARTIFACTS <<< "$OPTARG"
+      ;;
+    r)
+      REPO="$OPTARG"
+      ;;
+    w)
+      WORKFLOW="$OPTARG"
+      ;;
+    b)
+      BRANCH="$OPTARG"
+      ;;
+    i)
+      WORKFLOW_RUN_ID="$OPTARG"
+      ;;
+    d)
+      TARGET_DIR="$OPTARG"
+      ;;
+    z)
+      DECOMPRESS="true"
+      ;;
+    t)
+      TOKEN="$OPTARG"
+      ;;
+    ?)
+      usage
+      exit 1
+      ;;
+  esac
+done
+
+if [[ -z "${TARGET_DIR}" ]]; then
+  echoerr target directory must be specified.
+  usage
+  exit 1
+fi
+
+if [[ -z "${WORKFLOW_RUN_ID}" ]]; then
+  if [[ -z "${BRANCH}" ]] || [[ -z "${WORKFLOW}" ]]; then
+    echoerr 'You must either specify a workflow_run_id (-i), or a branch (-b) and a workflow file (-w).'
+    usage
+    exit 1
+  fi
+fi
+
+curl_attri=("--retry" "3")
+curl_attri+=("--silent")
+curl_attri+=("--show-error")
+curl_attri+=('-H' 'Accept: application/vnd.github.v3+json')
```

Comment:
This should all be single quotes I think, though it doesn't matter much.

Reply:
shellcheck thinks it's fine since it's parsed as an array.

Reply:
I suppose I could split them '-H' 'Accept: application/vnd.github.v3+json' ?

Reply:
Hmm I'm not sure what you mean?